File: go/src/leetcode/editor/cn/0243_shortest_word_distance.go
package main

import "math"

func main() {
	println("243、shortest_word_distance")
}

// leetcode submit region begin(Prohibit modification and deletion)
func shortestDistance(wordsDict []string, word1 string, word2 string) int {
	l, r, e := -1, -1, len(wordsDict)
	for i, v := range wordsDict {
		if word1 == v {
			l = i
		}
		if word2 == v {
			r = i
		}
		if l > -1 && r > -1 {
			e = int(math.Min(float64(e), math.Abs(float64(l-r))))
		}
	}
	return e
}

//leetcode submit region end(Prohibit modification and deletion)

File: rust/src/leetcode/editor/cn/_0243_shortest_word_distance.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("243、shortest_word_distance");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    #[allow(dead_code)]
    pub fn shortest_distance(words_dict: Vec<String>, word1: String, word2: String) -> i32 {
        let (mut res, mut left, mut right) = (words_dict.len() as i32, -1, -1);
        for i in 0..words_dict.len() {
            if word1 == words_dict[i as usize] {
                left = i as i32;
            }
            if word2 == words_dict[i as usize] {
                right = i as i32;
            }
            if left > -1 && right > -1 {
                res = ((right - left) as i32).abs().min(res)
            }
        }
        res
    }
}
//leetcode submit region end(Prohibit modification and deletion)
